spec/models/profile_spec.rb

```diff
@@ -333,7 +362,38 @@
       )
       allow_any_instance_of(UserMailer).to receive(:params).and_return(fake_params)
 
-      expect { profile }.to change(ActionMailer::Base.deliveries, :count).by(1)
+      expect { profile.reject_for_fraud }.to change(ActionMailer::Base.deliveries, :count).by(1)
+    end
+
+    context 'when the SP is the IRS' do
+      it 'logs an event' do
+        sp = create(:service_provider, :irs)
+        profile = create(
+          :profile, user: user, active: false, fraud_review_pending: true,
+                    initiating_service_provider: sp
+        )
+        allow(IdentityConfig.store).to receive(:irs_attempt_api_enabled).and_return(true)
+        expect(profile.initiating_service_provider.irs_attempts_api_enabled?).to be_truthy
+
+        expect_any_instance_of(IrsAttemptsApi::Tracker).to receive(:fraud_review_adjudicated).
+          with(determination: 'rejected')
+
+        profile.reject_for_fraud
+      end
+    end
+
+    context 'when the SP is not hte IRS' do
+      it 'does not log an event' do
+        sp = create(:service_provider)
+        profile = create(
+          :profile, user: user, active: false, fraud_review_pending: true,
+                    initiating_service_provider: sp
```

Comment:
one rule of thumb I like for arguments is "all same line or all different lines" (and same for all the other examples)

```suggestion
          :profile,
          user: user,
          active: false,
          fraud_review_pending: true,
          initiating_service_provider: sp
```